pkg/k8shandler/cluster.go

```diff
@@ -222,42 +224,46 @@ func (er *ElasticsearchRequest) setUUIDs() {
 				continue
 			}
 
-			// update the node to set uuid
-			cluster.Spec.Nodes[index].GenUUID = &uuid
+			er.setUUID(index, uuid)
+		}
+	}
+}
 
-			nretries := -1
-			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				nretries++
-				if err := er.client.Get(context.TODO(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, cluster); err != nil {
-					// FIXME: return structured error
-					log.Info("Could not get Elasticsearch cluster", "cluster", cluster.Name, "error", err)
-					return err
-				}
+func (er *ElasticsearchRequest) setUUID(index int, uuid string) {
+	nretries := -1
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		nretries++
+		if err := er.client.Get(context.TODO(), types.NamespacedName{Name: er.cluster.Name, Namespace: er.cluster.Namespace}, er.cluster); err != nil {
+			// FIXME: return structured error
+			log.Info("Could not get Elasticsearch cluster", "cluster", er.cluster.Name, "error", err)
```

Comment:
We should log the NS since it can watch many

Reply:
i'm just changing indentation here but i can expand on it as well